### ISSUE TITLE
Flaky TestRunCronExample

### ISSUE THREAD

**Issue by squakez** [apache/camel-k#5533]
```
❌ TestRunCronExample (10m59.86s)
      cron_test.go:54: 
          Timed out after 420.000s.
          Expected
              <string>: 
          to contain substring
              <string>: Magicstring!
```
Last time seen in https://github.com/apache/camel-k/actions/runs/9184516211/job/25256969729#step:4:12970

**Comment by squakez** [comment]
Still relevant :(

**Comment by gansheer** [comment]
Looks like we are good now.

**Comment by squakez** [comment]
@gansheer would you mind backporting all the commits that fixed this into `release-2.4.x`? Thanks.
